tests/integration/test_charm.py

```diff
@@ -23,24 +25,45 @@
 async def test_build_and_deploy(
     ops_test: OpsTest,
     charm_base: str,
-    filesystem_client_charm,
-    nfs_server_proxy_charm,
-    cephfs_server_proxy_charm,
+    filesystem_client_charm: Awaitable[str | pathlib.Path],
+    nfs_server_proxy_charm: Awaitable[str | pathlib.Path],
+    cephfs_server_proxy_charm: Awaitable[str | pathlib.Path],
 ) -> None:
     """Build the charm-under-test and deploy it together with related charms.
 
     Assert on the unit status before any relations/configurations take place.
     """
-    nfs_info = bootstrap_nfs_server()
-    cephfs_info = bootstrap_microceph()
-    logger.info(f"Deploying {', '.join(CHARMS)}")
 
-    # Pack charms.
-    filesystem_client, nfs_server_proxy, cephfs_server_proxy = await asyncio.gather(
-        filesystem_client_charm, nfs_server_proxy_charm, cephfs_server_proxy_charm
-    )
+    async def deploy_cephfs_proxy():
+        """Deploy a MicroCeph cluster and pass its info to a new `cephfs_server_proxy_charm`."""
+        cephfs_server_proxy = await cephfs_server_proxy_charm
+        (cephfs_info, _) = await asyncio.gather(
+            bootstrap_microceph(ops_test),
+            ops_test.model.deploy(str(cephfs_server_proxy), application_name=CEPHFS_SERVER_PROXY),
+        )
+        await ops_test.model.applications[CEPHFS_SERVER_PROXY].set_config(
+            {
+                "fsid": cephfs_info.fsid,
+                "sharepoint": f"{cephfs_info.name}:{cephfs_info.path}",
+                "monitor-hosts": " ".join(cephfs_info.monitor_hosts),
+                "auth-info": f"{cephfs_info.user}:{cephfs_info.key}",
+            }
+        )
+
+    async def deploy_nfs_proxy():
+        """Deploy an NFS server and pass its info to a new `nfs_server_proxy_charm`."""
+        nfs_server_proxy = await nfs_server_proxy_charm
+        (nfs_info, _) = await asyncio.gather(
```

Comment:
Same thing as another one of my comments. Are these parenthesis necessary to guarantee the result of the Future? 

```suggestion
        nfs_info, _ = await asyncio.gather(
```